guides/source/security.md

```diff
@@ -1140,7 +1142,11 @@ class PostsController < ApplicationController
 end
 ```
 
-You can enable automatic nonce generation:
+#### Adding a Nonce
+
+Disabling all inline JavaScript and CSS is recommended in a Content Security
+Policy. If you absolutely must use inline script tags, you can enable automatic
+nonce generation:
```

Comment:
Should we also include a "why" here?  Something like:

```suggestion
As part of defense-in-depth against XSS and injection attacks, it is recommended 
to define a Content Security Policy that disables inline JavaScript and CSS. But 
if you must use inline script tags, you can configure a nonce generator:
```

Reply:
Yes, I was thinking about that as well.
I think that would be better at the beginning of the Content Security Policy,
as an introduction.

Reply:
I've added it to under the `Content Security Policy` heading, and reworded it a bit.
Let me know what you think.

Reply:
Sure, that looks good.  Although "Disabling all inline JavaScript and CSS is recommended in a Content Security
Policy" still sounds a little awkward to my ears.  (Maybe it's just me?)

Also, "If you absolutely must" sounds a bit harsh, as if using inline scripts is bad practice.  It is perhaps not best practice, but, as long as it is secure, I don't think we should imply that it is bad practice.

Reply:
I'll try rewording that part 🙂 .

> Also, "If you absolutely must" sounds a bit harsh, as if using inline scripts is bad practice. It is perhaps not best practice, but, as long as it is secure, I don't think we should imply that it is bad practice.

Another example which also uses "absolutelty", just like the google page I linked to:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src#unsafe_inline_script

Maybe it's too harshly worded, but I'd prefer too make clear using a nonce has drawbacks,
and the readers are aware of these drawbacks.
Maybe we can link to the following in a warning block?

https://www.w3.org/TR/CSP3/#security-nonces
> Note: Using a nonce to allow inline script or style is less secure than not using a nonce, as nonces override the restrictions in the directive in which they are present. An attacker who can gain access to the nonce can execute whatever script they like, whenever they like. That said, nonces provide a substantial improvement over ['unsafe-inline'](https://www.w3.org/TR/CSP3/#grammardef-unsafe-inline) when layering a content security policy on top of old code. When considering ['unsafe-inline'](https://www.w3.org/TR/CSP3/#grammardef-unsafe-inline), authors are encouraged to consider nonces (or hashes) instead.

Reply:
BTW that Google article is a lot harsher:
> You can also use a nonce or a hash (see below), but you really shouldn't.

Reply:
> Makes you wonder if everyone just copies each others content :laughing: .

Indeed!

> Maybe it's too harshly worded, but I'd prefer too make clear using a nonce has drawbacks, and the readers are aware of these drawbacks. Maybe we can link to the following in a warning block?
> 
> https://www.w3.org/TR/CSP3/#security-nonces
> 
> > Note: Using a nonce to allow inline script or style is less secure than not using a nonce, as nonces override the restrictions in the directive in which they are present. An attacker who can gain access to the nonce can execute whatever script they like, whenever they like. That said, nonces provide a substantial improvement over ['unsafe-inline'](https://www.w3.org/TR/CSP3/#grammardef-unsafe-inline) when layering a content security policy on top of old code. When considering ['unsafe-inline'](https://www.w3.org/TR/CSP3/#grammardef-unsafe-inline), authors are encouraged to consider nonces (or hashes) instead.

That's a good explanation.  Yeah, let's link to that.  And maybe *copy* some of it too... :stuck_out_tongue_winking_eye:

Reply:
Maybe something like
```suggestion
Nonces provide a substantial improvement over 'unsafe-inline' when implementing a Content Security Policy on top existing code. If you are considering 'unsafe-inline' consider using nonces instead. 
```

Reply:
@jonathanhefner I've change the text and added link.